engine/src/main/java/org/hibernate/search/backend/IndexWorkVisitor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend;
+
+/**
+ * Contract for visitors of {@link LuceneWork} types.
+ * <p>
+ * Implementations can receive a specific input parameter and return a specific output type. Use {@link Void} as
+ * parameterization for each of the two not needed.
+ *
+ * @author Sanne Grinovero
+ * @author Gunnar Morling
+ * @param <P> Context parameter type expected by a specific visitor
+ * @param <R> Return type provided by a specific visitor
+ */
+public interface IndexWorkVisitor<P, R> {
```

Comment:
What about my comment that this interface cab be simplified if one just slightly changes the code in `NRTWorkspaceImpl`?